.travis.yml

```diff
@@ -14,68 +14,142 @@ env:
     - BUILD_NUMBER=$TRAVIS_BUILD_NUMBER
     - BUILD_VERSION_MAJOR=0
 
+before_install:
+  - export BUILD_VERSION=${TRAVIS_TAG:-"$BUILD_VERSION_MAJOR-dev.$TRAVIS_BUILD_NUMBER"}
+
 cache:
   directories:
     - "$CACHE_ROOT_DIR"
+    - vendor
 
-install:
-  - source bin/travis_scripts/setup_tools_dir.sh $BUILD_TOOLS_PATH
-  - source bin/travis_scripts/ensure_glide.sh $BUILD_TOOLS_PATH "v0.13.1"
-  - bin/travis_scripts/ensure_xgo.sh
-  - glide "-home" $GLIDE_HOME install
+stages:
+  - dep-cache
+  - test
+  - name: e2e-test
+    if: type = pull_request
+  - name: build
+    if: (type != pull_request AND branch = master AND tag IS blank) OR tag IS present
+  - name: release-dev
+    if: type != pull_request AND branch = master AND tag IS blank
+  - name: release
+    if: tag IS present
+
+jobs:
+  include:
+    # Dependencies cache stage
+    - stage: dep-cache
+      name: "Vendor update"
+      script:
+        - source bin/travis_scripts/ensure_glide.sh $BUILD_TOOLS_PATH "v0.13.1"
+        - glide "-home" $GLIDE_HOME install
+
+    # Test stage
+    - stage: test
+      script: bin/check_vet
+      name: "Vet check"
+
+    - script: bin/builder_run bin/check_lint
+      name: "Linter"
+
+    - script: bin/check_fmt
+      name: "Formating check"
+
+    - script: bin/check_license
+      name: "License check"
```

Comment:
These sanity checkers like license, linter, vet are very small jobs (few sec), is there a point to paralelize them - overhead of single job on travis is around 30 secs, and there is limited number of parallel jobs.

Reply:
I did a test, the first test was with license+vet+fmt in one job, and the second test with all steps in separate jobs. Got the same results for the whole stage: 1:54 for both cases.
If we are starting all jobs in parallel, the time of the stage will be equal to the longest job. 
We still need to wait for other jobs to finish, like unit tests. 
And, I think, separated jobs looks more clear in the Travis CI UI.

Reply:
💯  karma points for making tests and measurements. And yes - I was thinking between clear separations and a bit of optimisation. Looks good to me now